Truncation of long title strings

lzach83 opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Link to Issue
Closes: #5272 

## Description of Changes
- Strings that are longer than 120 characters get truncated with an ellipses after. 

## "How We Fixed It"
<!-- Brief description of solution, if bug, to be added once issue is resolved. -->
- Built a function to take in account the width of the device and the max allowed characters and truncate accorded to truthy values.

## Test Plan
- Find longer titles and make sure it truncates correctly. 

## Other Considerations
<!--- Follow-up tickets, breaking changes, etc -->
loom: https://www.loom.com/share/634359c395314910b9cf256b969f5df9?sid=e26203a0-5e73-4f62-bf39-d06bbf3bf0a8

masvelio commented on packages/commonwealth/client/scripts/views/components/component_kit/CWContentPage/CWContentPage.tsx:
```diff
@@ -128,6 +128,22 @@ export const CWContentPage = ({
+    const maxWidth = 0.5 * availableWidth;
+
+    if (str.length > 140 || availableWidth < maxWidth) {
+      const ellipsisWidth = '...'.length * 4;
```
what 4 means?

masvelio commented on packages/commonwealth/client/scripts/views/components/component_kit/CWContentPage/CWContentPage.tsx:
```diff
@@ -128,6 +128,22 @@ export const CWContentPage = ({
+
+    if (str.length > 140 || availableWidth < maxWidth) {
+      const ellipsisWidth = '...'.length * 4;
+      const truncatedLength = Math.floor((maxWidth - ellipsisWidth) / 8);
```
what 8 means?

masvelio reviewed (approved):
1. use meaningful const names instead of intiger number
2. move this out from component to some util file

lzach83 commented:
Added a bunch of comments and added types to the function. I extracted `truncate` all the way out to the `helpers` dir because it will also be used in #4985
